core/shared/src/main/scala/scalaz/zio/TaskR.scala

```diff
@@ -0,0 +1,566 @@
+package scalaz.zio
+
+import scalaz.zio.Exit.Cause
+import scalaz.zio.internal.{ Executor, Platform }
+
+import scala.concurrent.ExecutionContext
+
+/*
+ * PHASE 1 DONE
+ * PHASE 2 DONE
+ * PHASE 3 DONE
+ * PHASE 4 DONE
+ * PHASE 6 DONE
+ */
+
+/*
+ * EXTENDS ZIO_E_Throwable
+ */
+object TaskR {
+
+  type TaskrE = Throwable
+  // type TaskrR = Polymorphic
+
+  def apply[A](a: => A): Task[A] = effect(a)
+
+  /*
+   * ----------------------------------------------------------------------------------------------------------------------
+   * The Following is taken from ZIOFunctions.
+   * ----------------------------------------------------------------------------------------------------------------------
+   */
+
+  /** Returns an effect that models failure with the specified error.
+   * The moral equivalent of `throw` for pure code.
+   */
+  final def fail(error: TaskrE): IO[TaskrE, Nothing] = ZIO.fail(error)
+
+  /** Returns an effect that models failure with the specified `Cause`.
+   */
+  final def halt(cause: Cause[TaskrE]): IO[TaskrE, Nothing] = ZIO.halt(cause)
+
+  /** Returns an effect that models success with the specified strictly-
+   * evaluated value.
+   */
+  final def succeed[A](a: A): UIO[A] = ZIO.succeed(a)
+
+  /** Returns an effect that models success with the specified lazily-evaluated
+   * value. This method should not be used to capture effects. See
+   * `[[ZIO.effectTotal]]` for capturing total effects, and `[[ZIO.effect]]` for capturing
+   * partial effects.
+   */
+  final def succeedLazy[A](a: => A): UIO[A] = ZIO.succeedLazy(a)
+
+  /** Accesses the whole environment of the effect.
+   */
+  final def environment[R]: ZIO[R, Nothing, R] = ZIO.access(ZIO.identityFn[R])
```

Comment:
We don't really have a type alias for the combination but I see it a lot... we can fix it later.